kornia/morphology/morphology.py

```diff
@@ -151,10 +161,13 @@ def erosion(
         neighborhood = structuring_element.clone()
         neighborhood[kernel == 0] = -max_val
 
-    output = output.unfold(2, se_h, 1).unfold(3, se_w, 1)
-    output, _ = torch.min(output - neighborhood, 4)
-    output, _ = torch.min(output, 4)
-
+    B, C, H, W = tensor.size()
+    Hpad, Wpad = output.shape[-2:]
+    reshape_kernel = _neight2channels_like_kernel(kernel)
+    output, _ = F.conv2d(
+        output.view(B * C, 1, Hpad, Wpad), reshape_kernel, padding=0, bias=-neighborhood.view(-1).flip(0)
```

Comment:
erosion do not need to flip structuring element

Reply:
Hmmm, but the tests were not failed because of my flip. Could you please add one, which covers this?

Reply:
I am not able to PR my tests :(

Reply:
This is the new test for dilation:
```
    def test_flip(self, device, dtype):
        tensor = torch.tensor([[0.5, 1.0, 0.3], [0.7, 0.3, 0.8], [0.4, 0.9, 0.2]], device=device, dtype=dtype)[
            None, None, :, :
        ]
        kernel = torch.tensor([[0.0, 1.0, 1.0], [0.0, 1.0, 1.0], [0.0, 1.0, 1.0]], device=device, dtype=dtype)
        expected = torch.tensor([[0.7, 1.0, 1.0], [0.7, 1.0, 1.0], [0.7, 0.9, 0.9]], device=device, dtype=dtype)[
            None, None, :, :
        ]
        assert_allclose(dilation(tensor, kernel), expected)
```

This is the new test for erosion:
```
        def test_flip(self, device, dtype):
        tensor = torch.tensor([[0.5, 1.0, 0.3], [0.7, 0.3, 0.8], [0.4, 0.9, 0.2]], device=device, dtype=dtype)[
            None, None, :, :
        ]
        kernel = torch.tensor([[0.0, 1.0, 1.0], [0.0, 1.0, 1.0], [0.0, 1.0, 1.0]], device=device, dtype=dtype)
        expected = torch.tensor([[0.3, 0.3, 0.3], [0.3, 0.2, 0.2], [0.3, 0.2, 0.2]], device=device, dtype=dtype)[
            None, None, :, :
        ]
        assert_allclose(dilation(tensor, kernel), expected)
```

Reply:
Thank you!